### ISSUE TITLE
Create a high priority condition for alert rules

### ISSUE THREAD

**Issue by ceorourke** [getsentry/sentry#59872]
Create a new alert rule condition for "high priority" issues - we want a condition called "when a new high priority issue occurs" that will fire on any of:
* when a new issue with high severity is received
    * "high severity" is currently defined as a severity score higher than 0.1, but you might want to double check with Tillman and Jenn to see if that changed
* when an issue escalates for any reason
    * archived -> escalating
    * new low severity issue escalates

See more technical details in the tech spec [here](https://www.notion.so/sentry/Severity-based-Notifications-Tech-Spec-df83dc1696424075bd3a0ba2e565d916?pvs=4#5e21777509004f1db00e97dfa1f483e5) - we already have a filter and a condition (a filter is the "IF" and a condition is the "WHEN" in alert rule creation) for 2/3 of these so you can copy the logic, but you'll need to write a new one for when a new low severity issue escalates. 

Also worth noting that the archive until escalation condition uses [this](https://github.com/getsentry/sentry/blob/master/static/app/views/alerts/rules/issue/ruleNode.tsx#L319-L324) to override the text - I believe Nisanthan plans on removing the feature flags, but be aware of this when looking at the existing code.

This should be behind a separate feature flag from the existing one (`projects:first-event-severity-alerting`). The existing one is managed in [getsentry](https://github.com/getsentry/getsentry/blob/master/getsentry/features.py#L1631-L1638) and flags in specific projects . 

I anticipate questions from customers about why exactly it fired (or did not fire) so if possible, we should include the reason somewhere in the notification.

This is blocked by #60413, though if you really want to get started you could implement 2/3 of the condition beforehand.
